### PR TITLE
feat: targetframework to netstandard2.0

### PR DESCRIPTION
**Pull request by ByronMayne**
## About the PR
Changed the project to target `.netstandard2.0` instead of  `.net6.0`

### CODE REVIEW

File: test/LEGO.AsyncAPI.Tests/Writers/SpecialCharacterStringExtensionsTests.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            => this.Compose("null", "'null'");
+
+        [Test]
+        public void GetYamlCompatibleString_TildaWordString_ReturnsWrappedValue()
+              => this.Compose("~", "'~'");
+
+        // [Test]
+        // public void GetYamlCompatibleString_IntegerWithTwoPeriods_RendersPlainStyle()
+        //    => this.Compose("1.2.3", "1.2.3"); 
+        // TODO: This actually wraps in quotes which is wrong but this is already existing behaviour
```
Comment (ByronMayne) [inline test/LEGO.AsyncAPI.Tests/Writers/SpecialCharacterStringExtensionsTests.cs:30]: This actually wraps it in a quote, which is incorrect. This is because `1.2.3` is a valid datetime

File: src/LEGO.AsyncAPI/Writers/SpecialCharacterStringExtensions.cs [hunk 2]
```diff
@@ -183,11 +191,30 @@ internal static string GetYamlCompatibleString(this string input)
                 return $"'{input}'";
             }
 
-            // If string can be mistaken as a number, a boolean, or a timestamp,
-            // wrap it in quote to indicate that this is indeed a string, not a number, a boolean, or a timestamp
-            if (decimal.TryParse(input, NumberStyles.Float, CultureInfo.InvariantCulture, out var _) ||
-                bool.TryParse(input, out var _) ||
-                DateTime.TryParse(input, out var _))
+            if (DateTime.TryParse(input, out var _))
```
Comment (VisualBean) [inline src/LEGO.AsyncAPI/Writers/SpecialCharacterStringExtensions.cs:194]: are these changes needed for netstandard?

File: test/LEGO.AsyncAPI.Tests/Writers/SpecialCharacterStringExtensionsTests.cs (hunk 1 above)
Comment (VisualBean) [reply test/LEGO.AsyncAPI.Tests/Writers/SpecialCharacterStringExtensionsTests.cs:30]: remember that thing about dates and weird formats we talked about? 😅

**Comment by ByronMayne** [comment]
Everything should be good now

**Comment by VisualBean** [comment]
> Everything should be good now

looks like you kept the test changes around as well.

**Comment by ByronMayne** [comment]
Sorry about that, it was collapsed and I miss it in the visuals.

**Comment by VisualBean** [comment]
> Sorry about that, it was collapsed and I miss it in the visuals.

Took me a double take, to find it as well 😅 no worries! 
Thanks a lot for the contributions!
